src/ex4/client/src/App.jsx

```diff
@@ -0,0 +1,208 @@
+import "./App.css";
+import { useEffect, useState, useCallback, useMemo } from "react";
+import Toast from "monday-ui-react-core/dist/Toast";
+import ReactLoading from "react-loading";
+import InputComp from "./components/InputComp";
+import ItemsComp from "./components/ItemsComp";
+import ButtonComp from "./components/ButtonComp";
+import NumOfItemsComp from "./components/NumOfItemsComp";
+import clearIcon from "./images/clearIcon.png";
+import itemClient from "./item_client";
+
+function App() {
+	const [allItems, setAllItems] = useState([]);
+	const [lastPerformance, setLastPerformance] = useState({
+		item: {},
+		undoAction: "",
+		undoMessage: "",
+	});
+	const [isEmptyState, setIsEmptyState] = useState(true);
+	const [isloading, setIsLoading] = useState(false);
+	const [toast, setToast] = useState(false);
+	const [timeOutId, setTimeOutId] = useState(NaN);
+
+	const handleToastOpen = useCallback(() => {
+		let timeOut;
+		clearTimeout(timeOutId);
+		setToast(true);
+		timeOut = setTimeout(() => {
+			setToast(false);
+		}, 5000);
+		setTimeOutId(timeOut);
+	}, [timeOutId]);
+
+	const addItem = useCallback(
+		async (value) => {
+			setIsLoading(true);
+			const res = await itemClient.addItem({ itemName: value });
+			setIsLoading(false);
+			setIsEmptyState(false);
+			if (!res.isResOk) {
+				alert("this item alredy exist.");
+			} else {
+				setAllItems(res.items);
+				setLastPerformance({
+					item: res.items.pop(),
+					undoAction: "delete",
+					undoMessage: "Successfully created",
+				});
+				handleToastOpen();
+			}
+		},
+		[handleToastOpen]
+	);
+
+	const updateItem = useCallback(
+		async (item, itemId) => {
```

Comment:
the item already contains the ID, so it doesn't make sense to get also the id seperatly.
Unless you're separating it to itemId and itemUpdates which doesn't contains the ID